allensdk/brain_observatory/ecephys/ecephys_session.py

```diff
@@ -385,6 +443,55 @@ def get_pupil_data(self, suppress_pupil_data: bool = True) -> pd.DataFrame:
         """
         return self.api.get_pupil_data(suppress_pupil_data=suppress_pupil_data)
 
+    def _mask_invalid_stimulus_presentations(self, stimulus_presentations):
+        """Mask invalid stimulus presentations
+
+        Find stimulus presentations overlapping with invalid times
+        Mask stimulus names with "invalid_presentation", keep "start_time" and "stop_time", mask remaining data with np.nan
+
+        Parameters
+        ----------
+        stimulus_presentations : pd.DataFrame
+            table including all stimulus presentations
+
+        Returns
+        -------
+        pd.DataFrame :
+            table with masked invalid presentations
+
+        """
+
+        def overlap(a, b):
```

Comment:
could you move this utility to the bottom of the file?

Reply:
done